src/main/scala/magellan/mapreduce/DBReader.scala

```diff
@@ -93,6 +93,25 @@ private[magellan] class DBReader extends RecordReader[ShapeKey, MapWritable] {
             fld
           }
         }
+
+
+      case 'F' => {
+        if (decimalCount == 0) {
+          val b = Array.fill[Byte](length)(0)
+          dis.readFully(b)
+          val fld = new Text()
+          fld.set(new String(b))
+          fld
+        } else {
+          val b = Array.fill[Byte](length)(0)
+          dis.readFully(b)
+          val fld = new Text()
+          fld.clear()
+          fld.set(new String(b))
```

Comment:
1. do you need new String(b) here, i am not in front of a IDE but i thought there is a set method on text which accepts a byte array and start/ end positions?
2. do you need clear on a newly allocated text object?